apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -443,45 +448,176 @@ impl ProcessingState {
 }
 
 impl FetchDependencyGraphNodePath {
+    pub(crate) fn new(
+        schema: ValidFederationSchema,
+        type_conditioned_fetching_enabled: bool,
+        root_type: CompositeTypeDefinitionPosition,
+    ) -> Self {
+        let root_possible_types: Vec<_> = if type_conditioned_fetching_enabled {
+            schema.possible_runtime_types(root_type).unwrap()
```

Comment:
Todo: in general we don't want any panic ever, let's have all of these return Result<Self, _>